### ISSUE TITLE
Power usage of different items 

### ISSUE THREAD

**Issue by Klaranth** [pandorabox-io/in-game#220]
Hedgehog ; OgelGames ; SX ; jessicara ; Huhhila

Telemosaics should drain energy from the user. Use too many telemosaics in a row and you die of exhaustion.
This idea was part of the discussion earlier this morning in main with chuche952 who found it odd that the telemosaics consume no power whatsoever.
Which makes sense, everything on this server that works needs power one way or another. But not the telemosaics.

OgelGames : no.

Actually long ago something like this was suggested to reduce fast travel.
Don't remember anymore but BuckarooBanzai was one who talked about it, possibly behind original suggestion. Peanut might have something to do with it too. I do remember that there was some valid points but can't remember what / why.

A reason to consume your calcium

If going for all powered: Tubes could require air compressor but there should also be simple compressors that wont require technic network, probably just some kind of "powered tube" in network.

Fuel fired compressor?

Biofuel refineries can actually work without power depending on what you're processing, some materials probably do require either power or something.

We have different kind of compressors than air...

Autocrafters for sure would need power network connection being kind of robotic assembly stations.

Making more stuff require power does imply some logistic challenges for layouts... could require insulated lv/mv/hv digi cables.

Beacons also could require power but would be one of hardest change considering how those are used currently. Autocrafters would be in same category while not as hard as beacons. Tubes probably would be simplest (but still not simple).
What do you mean with insulated lv/mv/hv digi cables?
Similar to insulated digilines? Connect only to certain predefined directions?

Not sure yet, something like that.

For beacons it could be possible to allow technic power connection to access some upgrades, increased range and possibly longer beam. That would make it bit more manageable upgrade but still not too simple (simple as in world behavior change).
Actually what could be done with tubes: assume that machines do actually come with built in compressors to operate tubes. It is impossible to make it really realistic but having dynamic energy consumption based on tube lengths could kind of work.
And would also discourage very long tube networks.
That would actually keep it simple to build machine/tube networks but would still make it more interesting to try different setups and to try optimizations.

So if you dump say 12k beacons inside a mapchunk-sized area... and connect them to same power network, does that scale?

I don't think beacon count really matters there because with radius 50 effects 1 beacon (in theory) already can cover whole area, similar if you fill whole mapblock with just cables and switching stations: it wont change outcome in any way other than each station consuming little cpu for nodetimers.

Of course there's beacon specific things that affect outcome but those are there no matter if it is powered or not and therefore not really something that have to be considered.

Technic has pipeworks as hard dependency but still would be nice to minimize dependencies.

All MV recipes require pipeworks
Yeah, I think for actual basic use you wont need pipeworks for anything really.

Some existing builds likely expect machines not to pass power... almost makes me want a dedicated emergy killswitch node (behaviour to be defined)

Changing beacons to simply require power wont be good thing because of extremely wide current use in world, if anything it should be "soft" change which means allowing unpowered beacons to function somehow possibly just reducing some effects.
Some possibilities include: next activation halves beam length (full with power network), diamond block upgrades become optional (power network or diamond upgrade), some effects are only available when connected to power network (fly probably should not be changed).
Well, if fly range would be changed then probably radius >= 32 without power network and higher radius only with power network. Then while doing that fix effects when beacon becomes unloaded, that's because high ranges are buggy already so it kinda would allow changing it without that many unwanted side effects.
Basically simply having technic network connection should be able to fix unloaded area problem. Which should be fixed soon ...  Hopefully without actually loading those areas ...

**Comment by Klaranth** [comment]
knorke ; SX 
Has there been any discussions on having more usecases for powerusage (like the beacon)? In my experience, I usually have way more power, than I need.

**Comment by Klaranth** [comment]
baijin
Higher voltage cable should be able to handle lower voltages

1hit
Mixing voltages on machines and cable (except the supply converter, when used as intended), should result in: 1) destruction of the equipment, 2) fire, 3) electrocution of the player who rigged this up.
And a link to ElectroBoom's YouTube channel.

SX 
But... not sure if cable cross sections are backwards, looking at equipment it might be. Maybe it is just so much more insulation...

Naso 
I like this, now I have an excuse to make a small voltaic arc animated texture :D

**Comment by OgelGames** [comment]
> Changing beacons to simply require power wont be good thing because of extremely wide current use in world

Same goes for telemosaics.

**Comment by S-S-X** [comment]
> > Changing beacons to simply require power wont be good thing because of extremely wide current use in world
> 
> Same goes for telemosaics.

Reminder that this is not proposal here ^^.

As actual discussion is not very recent I wont reopen this yet but removing obscure label as there's not yet anything proposed to be fixed or changed but just discussion about many different possibilities.

**Comment by S-S-X** [comment]
@OgelGames  if you're confused about what is requested in this issue (if you added "confused" emoji as a reply because of that) then carefully reading through whole discussion should give you idea but basically:

Discussion is currently in pure brainstorming stage without any conclusions but with multiple different (possibly conflicting) ideas and approach proposals for those ideas.
Including at least telemosaics, technic machines, pipeworks tubes and autocrafters, beacons and biofuel refineries.

**Comment by Klaranth** [comment]
wf13
Idea: add this mod for simplicity of use with biofuel reactors: https://git.gundul.net/gundul42/fuelcontroller

**Comment by The4codeblocks** [comment]
just bringing this up here: power use could be an {up/side}grade
